### PR TITLE
completed object oriented programming

### PR DESCRIPTION
**Pull request by goelvatsal**

### CODE REVIEW

File: arrays/practiceObjectOriented/main.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package main
+
+type Rectangle struct {
+	length int
+	width  int
+}
+
+func Monolithic(length, width int) (int, int) {
+	var r Rectangle
+	r.initialize(length, width)
```
Comment (goel4607) [inline arrays/practiceObjectOriented/main.go]: shorten this as separate declaration is not required

File: arrays/practiceObjectOriented/main.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+	area := r.findArea()
+	peri := r.findPerimeter()
+	return area, peri
+}
+
+func (r *Rectangle) findArea() int {
```
Comment (goel4607) [inline arrays/practiceObjectOriented/main.go]: as discussed remove the pointers as it is not required

File: arrays/practiceObjectOriented/main.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	return r.length * r.width
+}
+
+func (r *Rectangle) findPerimeter() int {
```
Comment (goel4607) [inline arrays/practiceObjectOriented/main.go]: remove pointers

File: arrays/practiceObjectOriented/main.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	return 2 * (r.length + r.width)
+}
+
+func (r *Rectangle) initialize(length, width int) {
```
Comment (goel4607) [inline arrays/practiceObjectOriented/main.go]: I think you do not need separate initialize method as you are not doing anything special here

File: arrays/practiceObjectOriented/main_test.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package main
+
+import "testing"
+
+func TestMonolithic(t *testing.T) {
+	area, peri := Monolithic(10, 5)
```
Comment (goel4607) [inline arrays/practiceObjectOriented/main_test.go:6]: good to see that you are writing test cases 👍

**Review by goel4607 — CHANGES_REQUESTED** [review]

File: arrays/practiceObjectOriented/main.go (hunk 1 above)
Comment (goelvatsal) [reply arrays/practiceObjectOriented/main.go]: fixed

File: arrays/practiceObjectOriented/main.go (hunk 2 above)
Comment (goelvatsal) [reply arrays/practiceObjectOriented/main.go]: fixed

File: arrays/practiceObjectOriented/main.go (hunk 3 above)
Comment (goelvatsal) [reply arrays/practiceObjectOriented/main.go]: fixed

File: arrays/practiceObjectOriented/main.go (hunk 4 above)
Comment (goelvatsal) [reply arrays/practiceObjectOriented/main.go]: fixed

File: arrays/practiceObjectOriented/main_test.go (hunk 5 above)
Comment (goelvatsal) [reply arrays/practiceObjectOriented/main_test.go:6]: thank you

**Review by goel4607 — APPROVED** [review]

**Pull request merged**